lib/Sema/CSSolver.cpp

```diff
@@ -1000,7 +1002,7 @@ static PotentialBindings getPotentialBindings(ConstraintSystem &cs,
     }
 
     if (exactTypes.insert(type->getCanonicalType()).second)
-      addPotentialBinding({type, kind, None});
+      addPotentialBinding({type, kind, None}, /*allowJoinMeet=*/!adjustedIUO);
```

Comment:
Yeah, I felt like I wanted this to be super-IUO-specific. This goes away when we take IUOs out of the type system.